routing/router_test.go

```diff
@@ -263,7 +263,7 @@ func TestSendPaymentRouteFailureFallback(t *testing.T) {
 	var payHash [32]byte
 	paymentAmt := lnwire.NewMSatFromSatoshis(1000)
 	payment := LightningPayment{
-		Target:      ctx.aliases["luoji"],
+		Target:      ctx.aliases["sophon"],
```

Comment:
I am aware of these cases. But I still think we shouldn't rely on mission control to handle them. In my model of how this should work, such an attempt will fail and by the time we start another path finding round, the local channel data is updated and that channel won't be used anymore.

Reply:
Although this change is part of a commit that is now removed from the pr, I found a case where we are unfortunately still forced to penalize local channels.

When the error source is unknown, it may or may not be our peer. In that case we can only penalize all channels along the route including our own.